AI/BattleAI/BattleAI.cpp

```diff
@@ -530,8 +511,15 @@ void CBattleAI::attemptCastingSpell()
 
 	CStopWatch timer;
 
-	CThreadHelper threadHelper(&tasks, threadCount);
-	threadHelper.run();
+	std::vector<std::shared_ptr<void>> scriptsPool; //todo: re-implement scripts context cache

```

Comment:
In this case scripts can only modify corresponding `HypotheticBattle` that is thread local. Without cache each 'evaluateSpellcast' task will create fresh scripting context, the idea was to implement cache so it will one instance per thread but not per task.